draft-duke-quic-protected-initial.md

```diff
@@ -356,52 +350,55 @@ Its provisional type is 0x454348.
-to a failed attempt to use Protected Initials, and therefore only occurs in
-other versions of QUIC.
-
 If this transport parameter does not match this format, is received by a server,
-or is received in a connection that uses Protected Initials, the receiver MUST
-terminate the connection with a TRANSPORT_PARAMETER_ERROR.
+or is received in a connection where the client's most recent Initial had a
+non-zero-length Encryption Context, the receiver MUST terminate the connection
+with a TRANSPORT_PARAMETER_ERROR.
 
 # Intermediaries
 
 In the topology proposed in Section 3.1 of {{ECHO}}, where a client-facing
 server has its own public name and potentially fronts a number of independent
 domains, only the client-facing server has the private keys. Thus, it modifies
-incoming Initial Packets before forwarding to the back-end server. The back-end
-server uses a modified technique for decrypting Initial packets.
+incoming Initial Packets before forwarding to the back-end server.
 
 A common use case of this topology is a load balancer fronting multiple domains
 using the same IP address, which makes routing decisions based on the SNI in the
 Client Hello.
 
 ## Client-Facing Server
 
+The client-facing server is responsible for verifying the Initial packet is
+decryptable, and sending a 0-RTT packet (and dropping the Initial) if it is not.
+
 If an incoming client Initial has a non-zero length Encryption Context field,
-the client-facing server MUST rewrite the Config ID field as zero, replaces the
-Encapsulated Secret with the shared secret in plaintext, and updates the
-Encryption Context Length accordingly.
+the client-facing server MUST delete the Encryption Context field, and re-
+encrypt the packet with keys derived from the fallback salt.
 
-To avoid authentication failure, the client-facing server MUST re-encrypt the
-Initial packet with the new header as associated data.
+If the client is using the shared secret, the client-facing server MUST re-
+encrypt server Initial packets using keys derived from the shared secret.
 
-Non-Initial packets, as well as Initial packets that have a zero-length
-Encryption Context, pass unmodified through the client-facing server.
+Non-Initial packets pass unmodified through the client-facing server.
 
 Note that client-facing servers cannot inspect any packet payloads except for
 Initial packets.
 
 ## Back-End Server
 
-Back-end servers use the plaintext shared secret to generate Initial keys.
-
 Back-end servers MUST have an up-to-date copy of the ECHConfigList the client-
 facing server is using, though it need not hold the private key, in order to
-properly process the relevant transport parameters.
+properly process and generate the relevant transport parameters.
 
 In all other respects they operate as Protected Initial servers.
 
+# Version Negotiation
+
+Note that QUIC version 1 is not compatible with QUIC Protected Initials, as it
```

Comment:
I would simply remove this section. Versions are assumed incompatible by default and this doesn't provide anything useful?

Reply:
Opened #29